### DISCUSSION TITLE
How does YAMCS database system work.

### DISCUSSION THREAD

**Discussion by codenamefrance** [yamcs/yamcs#961] [Q&A] [answered]
Hi, I'm trying to find a way to implement some sort of built-in data storage, since I need to process some data and keep it in a database for a while; I'd want to do this inside YAMCS, using the already present database system; however I'm finding some difficulties in understanding how it works.
For example, I can't figure out what yarch is.

**Comment by xpromache** [comment, marked as answer]
yarch is some kind of abstraction built on top of rocskdb in order to make it look a bit more like a SQL database. Specifically it implements tables and some rudimentary sql select commands but there are no joins and many other things that a traditional database has. 

If you want to use it, you are in for a lot of pain. We make no guarantee that your code will work from one version to another.
I think it is better to use another database. What you can also use is raw RocksDB - there you have a (key, value) interface, maybe it's enough for your needs.

**Comment by codenamefrance** [comment]
Thanks a lot! I think I will follow your advice and just use another database for other stuff.
